internal/telemetry/client.go

```diff
@@ -280,7 +280,16 @@ func (c *client) Stop() {
 	// close request types have no body
 	r := c.newRequest(RequestTypeAppClosing)
 	c.scheduleSubmit(r)
-	c.flush()
+	if internal.BoolEnv("DD_CIVISIBILITY_ENABLED", false) {
+		cnn := make(chan struct{})
+		c.flushWithCallback(func() {
+			cnn <- struct{}{}
+			close(cnn)
+		})
+		<-cnn
+	} else {
+		c.flush()
+	}
```

Comment:
How about adding a telemetry client option usable in `ApplyOps` like `SyncFlushOnStop()` or equivalent?

Reply:
makes sense, I did the change in [9c8f9e7](https://github.com/DataDog/dd-trace-go/pull/2963/commits/9c8f9e7ce3754968b03f8fa750de3b137371a585)